src/ast.js

```diff
@@ -30,26 +30,31 @@ class Alternation {
  * Represents a character class (e.g. [0-9a-z])
  */
 class CharClass {
-  constructor(a, b) {
+  constructor(a, b, flags) {
     this.precedence = 1;
     this.set = regenerate(a, b);
+    this.flags = flags;
   }
 
   get length() {
     return 1;
   }
 
+  get isSingleCharacter() {
+    return !this.set.toArray().some(c => c > 0xffff);
```

Comment:
This should probably always return true if in unicode mode. Will get rid of the parentheses around the second character in your test case, which aren't necessary:

`\u261D(?:\u{1F3FB})?/u`